backend/question-service/src/models/question.repository.ts

```diff
@@ -18,6 +19,43 @@ export async function findOneQuestionByTitle(title: string): Promise<IQuestion |
     return questionModel.findOne({ title })
 }
 
+export async function findRandomQuestionByTopicAndComplexity(
+    category: Category,
+    complexity: Complexity
+): Promise<IQuestion | null> {
+    let complexityFilter
+    switch (complexity) {
+        case Complexity.EASY:
+            complexityFilter = `1${complexity}`
+            break
+        case Complexity.MEDIUM:
+            complexityFilter = `2${complexity}`
+            break
+        case Complexity.HARD:
+            complexityFilter = `3${complexity}`
+            break
+        default:
+            throw new Error('Invalid complexity')
+    }
```

Comment:
dont think this should be in repository file (it should perhaps be inside one of the types or model folders)

Reply:
okok can, ive created a file in types to handle this